extensions/sql/contract-definition/store/src/main/java/org/eclipse/dataspaceconnector/sql/contractdefinition/store/SqlContractDefinitionStore.java

```diff
@@ -0,0 +1,244 @@
+import org.eclipse.dataspaceconnector.sql.contractdefinition.schema.SqlContractDefinitionTables;
+import org.jetbrains.annotations.NotNull;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Stream;
+import javax.sql.DataSource;
+
+import static org.eclipse.dataspaceconnector.sql.SqlQueryExecutor.executeQuery;
+
+public class SqlContractDefinitionStore implements ContractDefinitionStore {
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private final DataSource dataSource;
+    private final TransactionContext transactionContext;
+
+    public SqlContractDefinitionStore(DataSource dataSource, TransactionContext transactionContext) {
+        this.dataSource = Objects.requireNonNull(dataSource);
+        this.transactionContext = Objects.requireNonNull(transactionContext);
+    }
+
+    @SuppressWarnings("unchecked")
+    private static <T> T readObject(String value) {
+        try {
+            Envelope envelope = OBJECT_MAPPER.readValue(value, Envelope.class);
+            return (T) OBJECT_MAPPER.readValue(envelope.getContent(), Class.forName(envelope.getClassName()));
+        } catch (Exception e) {
+            throw new EdcException(e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public @NotNull Collection<ContractDefinition> findAll() {
+        String query = "SELECT * from %s";
+
+        try (Connection connection = dataSource.getConnection()) {
+            return SqlQueryExecutor.executeQuery(
+                    connection,
+                    ContractDefinitionMapper.INSTANCE,
+                    String.format(query, SqlContractDefinitionTables.CONTRACT_DEFINITION_TABLE));
+        } catch (Exception exception) {
+            throw new EdcPersistenceException(exception);
+        }
+    }
+
+    @Override
+    public @NotNull Stream<ContractDefinition> findAll(QuerySpec spec) {
+        Objects.requireNonNull(spec);
+
+        Limit limit = Limit.Builder.newInstance()
+                .limit(spec.getLimit())
+                .offset(spec.getOffset())
+                .build();
+
+        String query = "SELECT * from %s " + limit.getStatement();
+
+        try (Connection connection = dataSource.getConnection()) {
+            List<ContractDefinition> definitions = SqlQueryExecutor.executeQuery(
+                    connection,
+                    ContractDefinitionMapper.INSTANCE,
+                    String.format(query, SqlContractDefinitionTables.CONTRACT_DEFINITION_TABLE));
+            return definitions.stream();
+        } catch (Exception exception) {
+            throw new EdcPersistenceException(exception);
+        }
+    }
+
+    @Override
+    public void save(Collection<ContractDefinition> definitions) {
+        Objects.requireNonNull(definitions);
+
+        String query = String.format("INSERT INTO %s (%s, %s, %s, %s) VALUES (?, ?, ?, ?)",
```

Comment:
Nit: Seems like refactoring the query into a constant somewhere will make it more impervious to refactoring, since you're using it here and in the `*loader`.